packages/backend/src/services/CoderService/CoderService.ts

```diff
@@ -0,0 +1,180 @@
+} from '@lightdash/common';
+import { ChartAsCode, currentVersion } from '@lightdash/common/src/types/coder';
+import { LightdashAnalytics } from '../../analytics/LightdashAnalytics';
+import { LightdashConfig } from '../../config/parseConfig';
+import { DashboardModel } from '../../models/DashboardModel/DashboardModel';
+import { ProjectModel } from '../../models/ProjectModel/ProjectModel';
+import { SavedChartModel } from '../../models/SavedChartModel';
+import { SpaceModel } from '../../models/SpaceModel';
+import { SchedulerClient } from '../../scheduler/SchedulerClient';
+import { BaseService } from '../BaseService';
+
+type CoderServiceArguments = {
+    lightdashConfig: LightdashConfig;
+    analytics: LightdashAnalytics;
+    projectModel: ProjectModel;
+    savedChartModel: SavedChartModel;
+    dashboardModel: DashboardModel;
+    spaceModel: SpaceModel;
+    schedulerClient: SchedulerClient;
+};
+
+export class CoderService extends BaseService {
+    lightdashConfig: LightdashConfig;
+
+    analytics: LightdashAnalytics;
+
+    projectModel: ProjectModel;
+
+    savedChartModel: SavedChartModel;
+
+    dashboardModel: DashboardModel;
+
+    spaceModel: SpaceModel;
+
+    schedulerClient: SchedulerClient;
+
+    constructor({
+        lightdashConfig,
+        analytics,
+        projectModel,
+        savedChartModel,
+        dashboardModel,
+        spaceModel,
+        schedulerClient,
+    }: CoderServiceArguments) {
+        super();
+        this.lightdashConfig = lightdashConfig;
+        this.analytics = analytics;
+        this.projectModel = projectModel;
+        this.savedChartModel = savedChartModel;
+        this.dashboardModel = dashboardModel;
+        this.spaceModel = spaceModel;
+        this.schedulerClient = schedulerClient;
+    }
+
+    private static transformChart(
+        chart: SavedChartDAO,
+        spaceSummary: Pick<SpaceSummary, 'uuid' | 'slug'>[],
+    ): ChartAsCode {
+        const spaceSlug = spaceSummary.find(
+            (space) => space.uuid === chart.spaceUuid,
+        )?.slug;
+        if (!spaceSlug) {
+            throw new NotFoundError(`Space ${chart.spaceUuid} not found`);
+        }
+
+        return {
+            name: chart.name,
+            description: chart.description,
+            tableName: chart.tableName,
+            updatedAt: chart.updatedAt,
+            metricQuery: chart.metricQuery,
+            chartConfig: chart.chartConfig,
+            dashboardUuid: chart.dashboardUuid,
+            colorPalette: chart.colorPalette,
+            slug: chart.slug,
+            tableConfig: chart.tableConfig,
+
+            spaceSlug,
+            version: currentVersion,
+        };
+    }
+
+    async getCharts(user: SessionUser, projectUuid: string) {
+        const project = await this.projectModel.get(projectUuid);
+        if (!project) {
+            throw new NotFoundError(`Project ${projectUuid} not found`);
+        }
+
+        // TODO handle permissions
```

Comment:
should add this before shipping there's already an endpoint that gets all charts for a project: https://github.com/lightdash/lightdash/blob/259bb325630bebcf8cd26b6cf6dffbc98e82afbb/packages/backend/src/services/ProjectService/ProjectService.ts#L3841-L3841

Reply:
Done, as Katie suggested, right now it is only avaialable for admins